app/policies/namespace_policy.rb

```diff
@@ -52,6 +52,16 @@ def create?
     (APP_CONFIG.enabled?("user_permission.create_namespace") || user.admin?) && push?
   end
 
+  def destroy?
```

Comment:
This is allowing any namespace to be destroyed if user is admin and delete is enabled. That is not true as described in destroy service object code.

Reply:
We need to fix this in the authorization level somehow to avoid showing the delete button when deletion would be denied. That's why I suggested it here.

Reply:
I'm not sure I'm following :confused: In order to show/hide the delete button, you need some code in the authorization level? Isn't it possible to get this right from a helper somehow?

Reply:
I do use a helper for that but I usually reuse the logic from the policies. That's what I've been doing for other cases. We've been strict in other policies rules but not in this one. What's the line between having stuff being done inside policies or not?

Reply:
Fair point. The distinction is one the semantics imho. It's not about "you are not authorized to perform this", rather "what you are trying to do is not possible because of reason X". In this case, it's not possible because we don't want to implement namespace removal until we don't implement user removal.